webapp-client/src/component/AddCategoryForm.js

```diff
@@ -1,32 +1,76 @@
 import React from 'react';
+import ErrorMessage from './ErrorMessage';
 
 export default class AddCategoryForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      hasError: false,
+    };
+    this.createCategory = this.createCategory.bind(this);
+    this.clearHasError = this.clearHasError.bind(this);
+  }
   // This function will be changed later to directly POST request to db instead of calling addCategory
-  createCategory = (e) => {
+  async createCategory(e) {
     e.preventDefault();
     const category = e.target.elements['add-category'].value;
-    this.props.addCategory(category);
-    e.target.elements['add-category'].value = ''; 
-  };
+    e.target.elements['add-category'].value = '';
+    const request = {
+      method: 'POST',
+      body: JSON.stringify({
+        title: category,
+        userId: 'fake-user-id',
+        items: [],
```

Comment:
You can delete the items: [] line. I think it'll just automatically assume that if u don't enter anything

Reply:
actually, is it better to include it so it is clearer what type of variable it is?

Reply:
nah it's okay, bc the type is indicated in the model.